app/models/post.rb

```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'Author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def counter
```

Comment:
### Optional
- [ ] It would be better if we use Rails Active Callbacks such as `after_save` on the counter method. We can also consider make the method private. For more insight kindly check this [link](https://guides.rubyonrails.org/active_record_callbacks.html)

Reply:
@addod19 Thank you very much for your review. When using after_save it is increasing the post count regardless the post it created or updated, This means that the post counter will be also increased when that post is commented or liked, because that post object is updating its properties values. It can be used for likes and comments but not for posts. 
In any case, thank you very much for this tip because I read the link you sent me and found after_create callback that I used for the counter method in the post model.